ansys_optical_automation/application/example_windshield_distortion_lpf_reader.py

```diff
@@ -0,0 +1,191 @@
+    data.error_manager(error)
+    distortion_analysis_info = []
+    if sequence:
+        for trace in traces:
+            lpf_in = vector_normalize(
+                [
+                    trace.vImpacts.Get(1).Get(0) - trace.vImpacts.Get(0).Get(0),
+                    trace.vImpacts.Get(1).Get(1) - trace.vImpacts.Get(0).Get(1),
+                    trace.vImpacts.Get(1).Get(2) - trace.vImpacts.Get(0).Get(2),
+                ]
+            )
+            lpf_out = [
+                trace.LastDirection.Get(0),
+                trace.LastDirection.Get(1),
+                math.sqrt(1 - trace.LastDirection.Get(0) ** 2 - trace.LastDirection.Get(1) ** 2),
+            ]
+            alpha = math.acos(vector_dot_product(lpf_in, lpf_out))
+            if lpf_out[1] < 0:
+                alpha = -alpha
+            else:
+                alpha = alpha
+            # print(alpha, lpf_out)
+            distortion_analysis_info.append(LpfRay(alpha, trace))
+    distortion_analysis_info = sorted(
+        distortion_analysis_info, key=lambda x: (x.start_position[0], x.start_position[1]), reverse=True
+    )
+    return distortion_analysis_info
+
+
+def add_lpf_data(data_list):
+    """
+    function to add lpf data for windshield distortion study.
+
+    Parameters
+    ----------
+    data_list : list
+        a list of ref_lpf_process: first as reference ray data followed by sampling ray data
+
+    Returns
+    -------
+
+
+    """
+    ref = data_list[0]
+    for data_idx, data in enumerate(data_list[1:]):
+        for item_idx, item in enumerate(ref):
+            ref[item_idx].rays.append(data[item_idx].rays[0])
+            ref[item_idx].calc_alpha.append(data[item_idx].ref_alpha)
+            if data_idx == len(data_list[1:]) - 1:
+                ref[item_idx].ECE_R43()
+    for item in ref:
+        print(item.start_position, item.ref_alpha, item.calc_alpha, item.optical_power)
+    return ref
+
+
+def plot_result(data_result):
+    """
+    function to plot result for windshield distortion data.
+
+    Parameters
+    ----------
+    data_result : list
+        a list of data in format: [x, y, optical power]
+
+    Returns
+    -------
+
+
+    """
+    x = []
+    y = []
+    value = []
+    for item in data_result:
+        x.append(item.start_position[0])
+        y.append(item.start_position[1])
+        if item.optical_power > 0.1:
+            value.append(0)
+        else:
+            value.append(item.optical_power)
+    plt.scatter(x, y, c=value, s=20)
+    plt.colorbar()
+    plt.show()
+
+
+def main():
+    """
+    main function to run windshield distortion study, available methods:
+        ECE_R43
+
+    Returns
+    -------
+
+
+    """
+    data = []
+    for selection in range(5):
+        file_name = getfilename("*.lpf")
+        if file_name == "":
+            print("Complete, no file has been selected")
+        else:
+            my_lpf = DpfLpfReader(231)
```

Comment:
don't hard code version number

Reply:
updated for users to select speos version